osquery/tables/system/darwin/cpu_time.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/*
+Inspired from the psutil per_cpu_times function -
+https://github.com/giampaolo/psutil/blob/ec1d35e41c288248818388830f0e4f98536b93e4/psutil/_psutil_osx.c#L739
+*/
+
+#include <mach/mach.h>
+#include <time.h>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+long int ticks_to_usecs(int ticks) {
+  return static_cast<long int>(
+      (static_cast<double>(ticks) / CLOCKS_PER_SEC * 1000000));
+}
+
+QueryData genCpuTime(QueryContext& context) {
+  QueryData results;
+
+  natural_t processor_count;
+  processor_cpu_load_info_data_t* processor_times;
+  mach_port_t host = mach_host_self();
+  mach_msg_type_number_t processor_msg_count;
+
+  kern_return_t ret =
+      host_processor_info(host,
+                          PROCESSOR_CPU_LOAD_INFO,
+                          &processor_count,
+                          reinterpret_cast<processor_info_t*>(&processor_times),
```

Comment:
We try to avoid `reinterpret_cast`s if at all possible, but I wasn't able to find a better path forward with `static_cast`s which is what we prefer. @theopolis do you by chance know of another way this could be written to leverage `static_cast` over `reinterpret`? I tried the following locally but that didn't seem to work out:
```
  processor_info_t* process_info;
...
  kern_return_t ret =
      host_processor_info(host,
                          PROCESSOR_CPU_LOAD_INFO,
                          &processor_count,
                          &process_info,
                          &processor_msg_count);
  auto processor_times = static_cast<processor_cpu_load_info_data_t*>(proc_info);
```